src/api/util/cache.ts

```diff
@@ -0,0 +1,82 @@
+import { FastifyReply, FastifyRequest } from 'fastify';
+import { logger } from '../../logger';
+
+/**
+ * A `Cache-Control` header used for re-validation based caching.
+ * * `public` == allow proxies/CDNs to cache as opposed to only local browsers.
+ * * `no-cache` == clients can cache a resource but should revalidate each time before using it.
+ * * `must-revalidate` == somewhat redundant directive to assert that cache must be revalidated, required by some CDNs
+ */
+const CACHE_CONTROL_MUST_REVALIDATE = 'public, no-cache, must-revalidate';
+
+export async function handleTokenCache(request: FastifyRequest, reply: FastifyReply) {
```

Comment:
(Maybe for a future PR) if some classes of tokens have specified TTLs, then even more efficient caching could be used (the "expires" based headers), which would allow browsers/CDNs to avoid a hitting the service with a request at all.

Reply:
Great idea, opened https://github.com/hirosystems/token-metadata-service/issues/66